From 2c353d086f7a18b2cff1f61ca4a63bac808eacb4 Mon Sep 17 00:00:00 2001
From: Wenlin Kang <wenlin.kang@windriver.com>
Date: Fri, 22 Jan 2021 07:35:11 +0000
Subject: [PATCH] staging: fsl-dpaa2/eth: remove __exit from
 dpaa2_eth_dbg_exit()

It fixes the follow warnings:

WARNING: modpost: Found 1 section mismatch(es).
To see full details build your kernel with:
'make CONFIG_DEBUG_SECTION_MISMATCH=y'

With 'make CONFIG_DEBUG_SECTION_MISMATCH=y', the details are reported:

WARNING: vmlinux.o(.init.text+0x4f6ec): Section mismatch in reference from the function dpaa2_eth_driver_init() to the function .exit.text:dpaa2_eth_dbg_exit()
The function __init dpaa2_eth_driver_init() references
a function __exit dpaa2_eth_dbg_exit().
This is often seen when error handling in the init function
uses functionality in the exit path.
The fix is often to remove the __exit annotation of
dpaa2_eth_dbg_exit() so it may be used outside an exit section.

The original patch was taken from NXP LSDK-18.09, it is slightly different with
the mainline https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
/commit/?id=091a19ea6e34bf5ee52b46e265ad244b029d74d2, here we are backporting
some of the the mainline changes to fix this issue.

Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-debugfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-debugfs.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-debugfs.c
index f4f69334a2f4..6af10721f335 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-debugfs.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-debugfs.c
@@ -349,7 +349,7 @@ void dpaa2_eth_dbg_init(void)
 	pr_info("DPAA2-ETH: debugfs created\n");
 }
 
-void __exit dpaa2_eth_dbg_exit(void)
+void dpaa2_eth_dbg_exit(void)
 {
 	debugfs_remove(dpaa2_dbg_root);
 }
-- 
2.26.1

